src/util/byte_operators.h

```diff
@@ -57,6 +57,13 @@ class byte_extract_exprt:public binary_exprt
   const exprt &offset() const { return op1(); }
 };
 
+template <>
+inline bool can_cast_expr<byte_extract_exprt>(const exprt &base)
+{
+  return base.id() == ID_byte_extract_little_endian ||
+         base.id() == ID_byte_extract_big_endian;
+}
+
```

Comment:
:pick: I guess we should probably implement `validate_expr` at the same time as `can_cast_expr`, for consistency with other classes which inherit from `binary_exprt`.
```
inline void validate_expr(const byte_extract_exprt &value)
{
  validate_operands(value, 2, "byte extract must have two operands");
}
```